docs(meshgateway): add example of unmanaged Deployment/Service for MeshGateway (#1740)
kumahq/kuma-website: app/_src/using-mesh/managing-ingress-traffic/builtin.md

guide but it's likely worth creating a `MeshGatewayInstance` to configure as
much as you can and using it as a basis for your resources.

Given a `MeshGateway` spec:

```yaml
spec:
  conf:
    listeners:
    - port: 80
      protocol: HTTP
  selectors:
  - match:
      kuma.io/service: demo-app_gateway
```

#### `Service`

The `Service` will forward traffic to the `kuma-dp` we'll configure in the next
section. Its `ports` need to be in sync with the `MeshGateway` `listeners`.

```yaml
metadata:
  annotations:
    kuma.io/gateway: builtin
  name: demo-app-gateway
  namespace: kuma-demo
spec:
  ports:
  - name: "80"
    port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: demo-app-gateway
```

The `selector` should match the `Pod` template created in the next section.

#### `Deployment`

The `Deployment` we'll create will manage running some number of `kuma-dp`
instances that are configured to serve traffic for your `MeshGateway`.

We'll cover just `spec.selector` and `spec.template` here because
other parts of the `Deployment` can be configured arbitrarily.
Most importantly you'll need to change:

* `kuma.io/tags` annotation - must match the `MeshGateway` `selectors`

> * `KUMA_CONTROL_PLANE_CA_CERT` environment variable - can be retrieved with `kubectl get secret kuma-tls-cert  -n kuma-system -o=jsonpath='{.data.ca\.crt}' | base64 -d`

Reviewer: I used `mesh_product_name_path` because I don't think we have something specific to "part of resource name" (?)

```suggestion
* `KUMA_CONTROL_PLANE_CA_CERT` environment variable - can be retrieved with `kubectl get secret {{site.mesh_product_name_path}}-tls-cert -n {{site.mesh_namespace}} -o=jsonpath='{.data.ca\.crt}' | base64 -d`
```

> * `containers[0].image` field - should be the version of Kuma you're using

Reviewer: ```suggestion
* `containers[0].image` field - should be the version of {{site.mesh_product_name}} you're using
```

            NFowEjEQMA4GA1UEAxMHa3VtYS1jYTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
            AQoCggEBALp6skN+nPPRQQ7Z2ZH4eDjcTtKnO/9n/ExM99tRuMgZCdaZe7zp5g6L
            1wvdgi3mtOlplplFaYvLlC9QPn6A7TQSCOPd28vhTIqoUhZ4V7yjr54h0bn6wmH+
            BdVXgnalXDb+mQtyDF4dvAY3f0QyOQAK3TjRp32OX+dYKsGGWtch1yiIPf7VnWPx
            4/K2v4DTjRuDcXg6S0x4GskGZ9zAhR1WJYEa/2uM+XBmy0GmF9INn0WOeje7Jf3c
            G5tCd+fPP6qEk5Q+tBVHd7Pz3xfD9TZjIfY1+h00UkpC50n/yE5FDK8aUpR17SUc
            QKJUskOivKAqRD1pI9zfhnHBJ2URfbcCAwEAAaNhMF8wDgYDVR0PAQH/BAQDAgKk
            MB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAPBgNVHRMBAf8EBTADAQH/
            MB0GA1UdDgQWBBQRT2hdjPpIt/FzcLJo/EWPFafCQzANBgkqhkiG9w0BAQsFAAOC
            AQEAO0Gfe750KCk+gMtBQHfHzEyQocO2qg2JXfZrBP/+rqeozbjXQj0BLYR9NXnp
            tLrxJHoBHdeE+TOnTFsxB7IRIkF1njEElKX4DVx7MjZCL1qLeWDuaXQmEgtFoWDM
            o4NqPZ4BIyuZZ9IVtYdeod5g5ucRopOP66zWr/RuwKFzdzni79BGaWuA3dNmLWcn
            MdsbZ165hdXstF6b48yPFKFrOdGgLpJheSCLlR/6vp5a+pA03fQZ6qV2j2uSqvAm
            XYl8Q3CdoI/yAX9p4mxkcYK7xaz0fQTrD/UyGD6l2cXgY90NP6LDAW1oYKHSD7Qc
            Y8vutIcexOtJfAdTZ3/GxBxH1Q==
            -----END CERTIFICATE-----
        - name: KUMA_CONTROL_PLANE_URL

>           value: https://kuma-control-plane.kuma-system:5678

Reviewer: ```suggestion
          value: https://{{site.mesh_cp_name}}.{{site.mesh_namespace}}:5678
```

        - name: KUMA_DATAPLANE_DRAIN_TIME
          value: 30s
        - name: KUMA_DATAPLANE_MESH
          value: default
        - name: KUMA_DATAPLANE_RUNTIME_TOKEN_PATH
          value: /var/run/secrets/kubernetes.io/serviceaccount/token
        - name: KUMA_DNS_CORE_DNS_BINARY_PATH
          value: coredns
        - name: KUMA_DNS_CORE_DNS_EMPTY_PORT
          value: "15054"
        - name: KUMA_DNS_CORE_DNS_PORT
          value: "15053"
        - name: KUMA_DNS_ENABLED
          value: "true"
        - name: KUMA_DNS_ENABLE_LOGGING
          value: "false"
        - name: KUMA_DNS_ENVOY_DNS_PORT
          value: "15055"
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: KUMA_DATAPLANE_RESOURCES_MAX_MEMORY_BYTES
          valueFrom:
            resourceFieldRef:
              containerName: kuma-gateway
              divisor: "0"
              resource: limits.memory

>         image: docker.io/kumahq/kuma-dp:2.6.2

Reviewer: need a new variable here like `{{site.mesh_docker_org}}`, also will `{{ page.version }}` work there or should it be something else?

```suggestion
        image: docker.io/{{site.mesh_docker_org}}/kuma-dp:{{page.latest_version}}
```

Author: added this new variable

        livenessProbe:
          failureThreshold: 12
          httpGet:
            path: /ready
            port: 9901
            scheme: HTTP
          initialDelaySeconds: 60
          periodSeconds: 5
          successThreshold: 1
          timeoutSeconds: 3

>         name: kuma-gateway

Reviewer: ```suggestion
        name: {{site.mesh_product_name_path}}-gateway
```

Author: I don't think we change the name of the container do we? :smile:

Reviewer: I just grepped the whole PR, but you're right in this case we don't
